frontend/www/js/omegaup/components/certificate/Mine.test.ts

```diff
@@ -0,0 +1,118 @@
+import { mount, shallowMount } from '@vue/test-utils';
+import { types } from '../../api_types';
+
+import T from '../../lang';
+import * as ui from '../../ui';
+
+import certificate_Mine from './Mine.vue';
+
+describe('Mine.vue', () => {
+  const propsData = {
+    certificates: [
+      {
+        certificate_type: 'contest',
+        date: new Date(),
+        name: 'Test contest',
+        verification_code: 'k2s8OPl620',
+      },
+      {
+        certificate_type: 'course',
+        date: new Date(),
+        name: 'Test course',
+        verification_code: '9soPa69jP3',
+      },
+      {
+        certificate_type: 'coder_of_the_month',
+        date: new Date(),
+        name: null,
+        verification_code: 'dodp874598',
+      },
+      {
+        certificate_type: 'coder_of_the_month_female',
+        date: new Date(),
+        name: null,
+        verification_code: 'gkspa12345',
+      },
+    ],
+    location: 'https://omegaup.com/',
+  };
+
+  it('Should handle an empty table', () => {
+    const wrapper = shallowMount(certificate_Mine, {
+      propsData: {
+        certificates: [] as types.CertificateListItem[],
+      },
+    });
+
+    expect(wrapper.text()).toContain(T.certificateListMineTitle);
+    expect(wrapper.find('table').exists()).toBeFalsy();
+    expect(wrapper.text()).toContain(T.certificateListMineCertificatesEmpty);
+  });
+
+  it('Should handle a table with data', () => {
+    const wrapper = mount(certificate_Mine, {
+      propsData,
+    });
+
+    expect(wrapper.text()).toContain(T.certificateListMineTitle);
+    expect(wrapper.find('table').exists()).toBeTruthy();
+    expect(wrapper.find('table tbody').text()).toContain(
+      ui.formatString(T.certificateListMineContest, {
+        contest_title: 'Test contest',
+      }),
+    );
+    expect(wrapper.find('table tbody').text()).toContain(
+      ui.formatString(T.certificateListMineCourse, {
+        course_name: 'Test course',
+      }),
+    );
+    expect(wrapper.find('table tbody').text()).toContain(
+      T.certificateListMineCoderOfTheMonth,
+    );
+    expect(wrapper.find('table tbody').text()).toContain(
+      T.certificateListMineCoderOfTheMonthFemale,
+    );
+  });
+
+  it('Should copy the verification code', () => {
+    const defineSpy = jest.spyOn(
+      (certificate_Mine as any).options.methods,
```

Comment:
Tratamos de evitar el uso del tipo `any`. Una de las formas que se me ocurre para arreglarlo es así:
```suggestion
      (certificate_Mine as unknown as Mine).options.methods,
```
para esto, tendrías que agregar una interfaz con que contenga los métodos y atributos que utilizamos en la prueba. Puede quedar algo así:

```ts

interface Mine {
  options: {
    methods: {
      getVerificationLink: (verificationCode: string) => string;
      getDownloadLink: (verificationCode: string) => string;
    }
  }
}
```